### PR TITLE
Added help print for demo script

### PR DESCRIPTION
**Pull request by siamaksade**

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/lgtm

File: demo [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 set -e -u -o pipefail
 
-declare -r SCRIPT_PATH=$(readlink -f "$0")
-declare -r SCRIPT_DIR=$(cd $(dirname "$SCRIPT_PATH") && pwd)
+declare -r SCRIPT_PATH=$(perl -e "print readlink '$0'") 2>/dev/null
```
Comment (sthaha) [inline demo]: why `perl` ( IMHO, unnecessary dependency) `bash` should be sufficient ? if you don't want to use `readlink` consider 

```
SCRIPT_DIR=$(cd -P $(dirname $0) && pwd)
```

File: demo [hunk 2]
```diff
@@ -81,9 +81,17 @@ demo.run() {
   demo.logs
 }
 
+demo.help() {
```
Comment (sthaha) [inline demo:83]: ❤️ 🤗

File: demo [hunk 3]
```diff
@@ -81,9 +81,17 @@ demo.run() {
...
+  echo "USAGE:"
```
Comment (sthaha) [inline demo]:
```
cat <<-EOF
help 
text the way you like 
it
EOF
```
may be better

File: demo [hunk 4]
```diff
@@ -81,9 +81,17 @@ demo.run() {
...
+  echo "demo prep           creates project and service account"
+  echo "demo setup          creates tasks, pipeline and resources"
+  echo "demo run            starts pipeline"
+}
+
 
 main() {
   local fn="demo.${1:-''}"
+  test "$fn" = "demo.''" && demo.help && exit 0
```
Comment (sthaha) [inline demo]: why do we need this? wouldn't it work without this line?

File: demo (hunk 1 above)
Comment (siamaksade) [reply demo]: 👍

File: demo (hunk 3 above)
Comment (siamaksade) [reply demo]: 👍

File: demo (hunk 4 above)
Comment (siamaksade) [reply demo]: Updated to remove that line

**Review by sthaha — APPROVED** [review]

**Pull request merged**